### PR TITLE
adding Try .NET for System.String.*

### PR DESCRIPTION
**Pull request by WilliamAntonRohm**
per https://dev.azure.com/mseng/TechnicalContent/_workitems/edit/1558837/

Sample code PR is https://github.com/dotnet/samples/pull/1149

This PR includes the following member groups and members:

- Compare
- Contains
- IndexOf
- IsNullOrEmpty
- Join
- Length
- Replace
- Split
- Substring
- Trim

### CODE REVIEW

**Comment by WilliamAntonRohm** [comment]
Triggering new build

**Comment by mairaw** [comment]
https://github.com/dotnet/samples/pull/1149 was merged now. @WilliamAntonRohm please see my merge comments there for some changes you'll have to make. 
Now that the samples PR is merged, I've triggered a build of the master branch of dotnet-api-docs to grab the latest samples version. Once that finishes building, we can reopen this PR to grab the latest version as well. Getting the dependency tree updated is a bit complicated.

**Comment by WilliamAntonRohm** [comment]
@mairaw -- Maira, I've removed "-interactive" for the 6 sample files where you undid changes.

File: xml/System/String.xml [hunk 1]
```diff
@@ -1735,10 +1737,13 @@ You can download the [Sorting Weight Tables](https://www.microsoft.com/download/
    
   
 ## Examples  
+
+ [!INCLUDE[interactive-note](~/includes/csharp-interactive-note.md)]
```
Comment (mairaw) [inline xml/System/String.xml]: You don't need this. You've already add it to the Remarks section.

```suggestion
```

File: xml/System/String.xml [hunk 2]
```diff
@@ -1735,10 +1737,13 @@ You can download the [Sorting Weight Tables](https://www.microsoft.com/download/
...
+
```
Comment (mairaw) [inline xml/System/String.xml]:
```suggestion
```

File: xml/System/String.xml [hunk 3]
```diff
@@ -4129,13 +4134,15 @@ You can download the [Sorting Weight Tables](https://www.microsoft.com/download/
  This method performs an ordinal (case-sensitive and culture-insensitive) comparison. The search begins at the first character position of this string and continues through the last character position.  
   
  To determine whether a string contains a specified substring by using something other than ordinal comparison (such as culture-sensitive comparison, or ordinal case-insensitive comparison), you can create a custom method. The following example illustrates one such approach. It defines a <xref:System.String> extension method  that includes a <xref:System.StringComparison> parameter and indicates whether a string contains a substring when using the specified form of string comparison.  
-  
+ 
+ [!INCLUDE[interactive-note](~/includes/csharp-interactive-note-some.md)]
```
Comment (mairaw) [inline xml/System/String.xml]: Move this to the examples section since it comes first in the page

File: xml/System/String.xml [hunk 4]
```diff
@@ -6628,7 +6635,7 @@ The `GetPinnableReference` method returns a character that can be used for pinni
  The following example searches for the "n" in "animal". Because string indexes begin at zero rather than one, the <xref:System.String.IndexOf%28System.String%29> method indicates that the "n" is at position 1.  
   
  [!code-cpp[System.String.IndexOf#12](~/samples/snippets/cpp/VS_Snippets_CLR_System/system.String.IndexOf/CPP/simple1.cpp#12)]
- [!code-csharp[System.String.IndexOf#12](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.IndexOf/CS/simple1.cs#12)]
+ [!code-csharp-interactive[System.String.IndexOf#12](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.IndexOf/CS/simple1.cs#12)]
```
Comment (mairaw) [inline xml/System/String.xml]: the note is not being displayed before this example on the page. The note should be added to the remarks of the MemberGroup element to show up high in the page.

File: xml/System/String.xml [hunk 5]
```diff
@@ -7984,12 +7991,13 @@ The `comparisonType` parameter is a <xref:System.StringComparison> enumeration m
   
  You can use the <xref:System.String.IsNullOrWhiteSpace%2A> method to test whether a string is `null`, its value is <xref:System.String.Empty?displayProperty=nameWithType>,  or it consists only of white-space characters.  
   
-## What is a null string?
+[!INCLUDE[interactive-note](~/includes/csharp-interactive-note-some.md)]
```
Comment (mairaw) [inline xml/System/String.xml]: all the samples for IsNullOrEmpty should be converted to interactive. Some need to move the snippet tag though to work. but then I'd move the note to the top of the examples section instead.

File: xml/System/String.xml [hunk 6]
```diff
@@ -7984,12 +7991,13 @@ The `comparisonType` parameter is a <xref:System.StringComparison> enumeration m
...
 
+## What is a null string?
```
Comment (mairaw) [inline xml/System/String.xml]:
```suggestion
## What is a null string?

```

File: xml/System/String.xml [hunk 7]
```diff
@@ -8217,7 +8225,9 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
  If `separator` is `null`, an empty string (<xref:System.String.Empty?displayProperty=nameWithType>) is used instead. If any member of `values` is `null`, an empty string is used instead.  
   
  <xref:System.String.Join%28System.String%2CSystem.Collections.Generic.IEnumerable%7BSystem.String%7D%29> is a convenience method that lets you concatenate each element in an `IEnumerable(Of String)` collection without first converting the elements to a string array. It is particularly useful with Language-Integrated Query (LINQ) query expressions. The following example passes a `List(Of String)` object that contains either the uppercase or lowercase letters of the alphabet to a lambda expression that selects letters that are equal to or greater than a particular letter (which, in the example, is "M"). The `IEnumerable(Of String)` collection returned by the <xref:System.Linq.Enumerable.Where%2A?displayProperty=nameWithType> method is passed to the <xref:System.String.Join%28System.String%2CSystem.Collections.Generic.IEnumerable%7BSystem.String%7D%29> method to display the result as a single string.  
-  
+
+[!INCLUDE[interactive-note](~/includes/csharp-interactive-note-some.md)]
```
Comment (mairaw) [inline xml/System/String.xml]: move the note to the membergroup remarks and the example for Join(String, String[], Int32, Int32) should also be made interactive

File: xml/System/String.xml [hunk 8]
```diff
@@ -9685,10 +9695,13 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
   
 ## Examples  
+
+[!INCLUDE[interactive-note](~/includes/csharp-interactive-note.md)]
+
  The following example demonstrates the <xref:System.String.Length%2A> property.  
   
  [!code-cpp[string.length#1](~/samples/snippets/cpp/VS_Snippets_CLR/string.length/CPP/length.cpp#1)]
- [!code-csharp[string.length#1](~/samples/snippets/csharp/VS_Snippets_CLR/string.length/CS/length.cs#1)]
+ [!code-csharp-interactive[string.length#1](~/samples/snippets/csharp/VS_Snippets_CLR/string.length/CS/length.cs#1)]
```
Comment (mairaw) [inline xml/System/String.xml]: the example from the remarks should also be interactive

File: xml/System/String.xml [hunk 9]
```diff
@@ -10567,7 +10580,9 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
   
  Because this method returns the modified string, you can chain together successive calls to the <xref:System.String.Replace%2A> method to perform multiple replacements on the original string. Method calls are executed from left to right. The following example provides an illustration.  
   
- [!code-csharp[System.String.Replace#2](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.string.replace/cs/replace2.cs#2)]
+[!INCLUDE[interactive-note](~/includes/csharp-interactive-note.md)]
```
Comment (mairaw) [inline xml/System/String.xml]: move the note to the remarks of the membergroup element

File: xml/System/String.xml [hunk 10]
```diff
@@ -10567,7 +10580,9 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
+
+ [!code-csharp-interactive[System.String.Replace#2](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.string.replace/cs/replace2.cs#2)]
```
Comment (mairaw) [inline xml/System/String.xml]: this sample still contains the full program. needs to be converted.

File: xml/System/String.xml [hunk 11]
```diff
@@ -10576,7 +10591,7 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
  The following example creates a comma separated value list by substituting commas for the blanks between a series of numbers.  
   
  [!code-cpp[string.replace1#1](~/samples/snippets/cpp/VS_Snippets_CLR/string.replace1/CPP/string.replace1.cpp#1)]
```
Comment (mairaw) [inline xml/System/String.xml]: the sample for Replace(String, String) should also be interactive

File: xml/System/String.xml [hunk 12]
```diff
@@ -10567,7 +10580,9 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
   
  Because this method returns the modified string, you can chain together successive calls to the <xref:System.String.Replace%2A> method to perform multiple replacements on the original string. Method calls are executed from left to right. The following example provides an illustration.
```
Comment (mairaw) [inline xml/System/String.xml]: this note with the interactive sample is repeated in both overloads. It would make more sense to move it to the remarks of the membergroup element as well and remove from the overloads.

File: xml/System/String.xml [hunk 13]
```diff
@@ -12005,17 +12029,18 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
   
  If `startIndex` is equal to zero, the method returns the original string unchanged.  
    
+[!INCLUDE[interactive-note](~/includes/csharp-interactive-note.md)]
```
Comment (mairaw) [inline xml/System/String.xml]: move the substring note to the remarks section of the membergroup element. and review the samples here. There are some missing interactivity.

File: xml/System/String.xml [hunk 14]
```diff
@@ -11629,6 +11650,9 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
    
   
 ## Examples  
+
+ [!INCLUDE[interactive-note](~/includes/csharp-interactive-note.md)]
```
Comment (mairaw) [inline xml/System/String.xml]: There's already a note about this in the remarks of the page

```suggestion
```

File: xml/System/String.xml [hunk 15]
```diff
@@ -11629,6 +11650,9 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
+
```
Comment (mairaw) [inline xml/System/String.xml]:
```suggestion
```

File: xml/System/String.xml [hunk 16]
```diff
@@ -14018,13 +14043,13 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
   
-   
+[!INCLUDE[interactive-note](~/includes/csharp-interactive-note-some.md)]  
   
 ## Examples  
  The following example uses the <xref:System.String.Trim?displayProperty=nameWithType> method to remove any extra white space from strings entered by the user before concatenating them.  
   
  [!code-cpp[System.String.Trim#2](~/samples/snippets/cpp/VS_Snippets_CLR_System/system.String.Trim/cpp/trim2.cpp#2)]
- [!code-csharp[System.String.Trim#2](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.Trim/cs/Trim2.cs#2)]
+ [!code-csharp-interactive[System.String.Trim#2](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.Trim/cs/Trim2.cs#2)]
```
Comment (mairaw) [inline xml/System/String.xml]: this doesn't seem to work well with try .net because it needs input and it doesn't seem to behave very well. I'd undo the change to this snippet (in the samples too)

File: xml/System/String.xml [hunk 17]
```diff
@@ -14018,13 +14043,13 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
   
  White-space characters are defined by the Unicode standard. The <xref:System.String.Trim> method removes any leading and trailing characters that produce a return value of `true` when they are passed to the <xref:System.Char.IsWhiteSpace%2A?displayProperty=nameWithType> method.  
   
-   
+[!INCLUDE[interactive-note](~/includes/csharp-interactive-note-some.md)]
```
Comment (mairaw) [inline xml/System/String.xml]: this note should be moved to the remarks section of the membergroup element

**Comment by WilliamAntonRohm** [comment]
@mairaw -- thank you for your review.  I'll make these changes shortly.

File: xml/System/String.xml (hunk 1 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done

File: xml/System/String.xml (hunk 3 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done

File: xml/System/String.xml (hunk 6 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done

File: xml/System/String.xml (hunk 2 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done

File: xml/System/String.xml (hunk 4 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done

File: xml/System/String.xml (hunk 13 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: I moved the note.  I looked at all the "Substring#" sample code references and they're all interactive.

File: xml/System/String.xml (hunk 14 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done

File: xml/System/String.xml (hunk 15 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done

File: xml/System/String.xml (hunk 16 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done, and snippet fixed in https://github.com/dotnet/samples/pull/1165

File: xml/System/String.xml (hunk 17 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

File: xml/System/String.xml (hunk 5 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done in https://github.com/dotnet/samples/pull/1171

File: xml/System/String.xml (hunk 7 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done in https://github.com/dotnet/samples/pull/1171

File: xml/System/String.xml (hunk 8 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done in https://github.com/dotnet/samples/pull/1171

File: xml/System/String.xml (hunk 9 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

File: xml/System/String.xml (hunk 10 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done in https://github.com/dotnet/samples/pull/1171

File: xml/System/String.xml (hunk 11 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done in https://github.com/dotnet/samples/pull/1171

Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done in https://github.com/dotnet/samples/pull/1171

File: xml/System/String.xml (hunk 12 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

**Comment by WilliamAntonRohm** [comment]
@mairaw -- I've addressed your review comments in String.xml here and in the [sample files PR](https://github.com/dotnet/samples/pull/1171)

File: xml/System/String.xml [hunk 18]
```diff
@@ -7979,32 +7993,35 @@ The `comparisonType` parameter is a <xref:System.StringComparison> enumeration m
  <xref:System.String.IsNullOrEmpty%2A> is a convenience method that enables you to simultaneously test whether a <xref:System.String> is `null` or its value is <xref:System.String.Empty?displayProperty=nameWithType>. It is equivalent to the following code:  
   
  [!code-cpp[System.String.IsNullOrEmpty#1](~/samples/snippets/cpp/VS_Snippets_CLR_System/system.string.isnullorempty/cpp/isnullorempty1.cpp#1)]
- [!code-csharp[System.String.IsNullOrEmpty#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.string.isnullorempty/cs/isnullorempty1.cs#1)]
+ [!code-csharp-interactive[System.String.IsNullOrEmpty#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.string.isnullorempty/cs/isnullorempty1.cs#1)]
```
Comment (mairaw) [inline xml/System/String.xml]: this small snippet shouldn't be interactive. just the others.

```suggestion
 [!code-csharp[System.String.IsNullOrEmpty#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.string.isnullorempty/cs/isnullorempty1.cs#1)]
```

**Comment by mairaw** [comment]
I kicked off a new build for the master branch which will allow me to pull the latest samples into this PR. I'll close and reopen this PR when that happens.

File: xml/System/String.xml (hunk 18 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

**Comment by mairaw** [comment]
Starting new build...

**Comment by mairaw** [comment]
Build has finished. Have you tested all your changes already @WilliamAntonRohm?
I'll start reviewing them as well.

File: xml/System/String.xml [hunk 19]
```diff
@@ -1616,7 +1618,7 @@ You can download the [Sorting Weight Tables](https://www.microsoft.com/download/
  The following example calls the <xref:System.String.Compare%28System.String%2CSystem.String%29> method to compare three sets of strings.
```
Comment (mairaw) [inline xml/System/String.xml]: There are still quite a few examples for Compare that could be made interactive:
https://review.docs.microsoft.com/en-us/dotnet/api/system.string.compare?view=netframework-4.8&branch=pr-en-us-2971

File: xml/System/String.xml [hunk 20]
```diff
@@ -2069,7 +2072,7 @@ You can download the [Sorting Weight Tables](https://www.microsoft.com/download/
  The following example compares two strings in three different ways: using linguistic comparison for the en-US culture; using linguistic case-sensitive comparison for the en-US culture; and using an ordinal comparison. It illustrates how the three methods of comparison produce three different results.  
   
  [!code-cpp[System.String.Compare#1](~/samples/snippets/cpp/VS_Snippets_CLR_System/system.String.Compare/cpp/example.cpp#1)]
- [!code-csharp[System.String.Compare#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.Compare/cs/Example.cs#1)]
+ [!code-csharp-interactive[System.String.Compare#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.Compare/cs/Example.cs#1)]
```
Comment (mairaw) [inline xml/System/String.xml]: we should undo this change to the sample as well. It relies on CultureInfo. 

One tip is to test out the inside of the Main method ahead of time on any page that has try .NET enabled and see if it would run and give you the same expected result.

File: xml/System/String.xml [hunk 21]
```diff
@@ -4129,24 +4132,27 @@ You can download the [Sorting Weight Tables](https://www.microsoft.com/download/
...
-  
+ 
  [!code-csharp[System.String.Contains#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/System.String.Contains/cs/ContainsExt1.cs#1)]
  [!code-vb[System.String.Contains#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/System.String.Contains/vb/ContainsExt1.vb#1)]  
   
  The following example then calls the `Contains` extension method to determine whether a substring is found in a string when using ordinal comparison and case-insensitive ordinal comparison.  
   
- [!code-csharp[System.String.Contains#2](~/samples/snippets/csharp/VS_Snippets_CLR_System/System.String.Contains/cs/ContainsExt1.cs#2)]
+ [!code-csharp-interactive[System.String.Contains#2](~/samples/snippets/csharp/VS_Snippets_CLR_System/System.String.Contains/cs/ContainsExt1.cs#2)]
```
Comment (mairaw) [inline xml/System/String.xml]: also need to undo this one here and on the samples 

```suggestion
 [!code-csharp[System.String.Contains#2](~/samples/snippets/csharp/VS_Snippets_CLR_System/System.String.Contains/cs/ContainsExt1.cs#2)]
```

File: xml/System/String.xml [hunk 22]
```diff
@@ -8082,6 +8099,15 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
       <Docs>
         <summary>Concatenates the elements of a specified array or the members of a collection, using the specified separator between each element or member.</summary>
         <altmember cref="M:System.String.Split(System.Char[])" />
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+  
+ [!INCLUDE[interactive-note](~/includes/csharp-interactive-note.md)]
```
Comment (mairaw) [inline xml/System/String.xml]:
```suggestion
 [!INCLUDE[interactive-note](~/includes/csharp-interactive-note-some.md)]
```

**Comment by WilliamAntonRohm** [comment]
@mairaw -- thank you for your review.  I am double-checking all the samples under these 10 member groups/members.

**Comment by mairaw** [comment]
Sounds good. I'm going through them as well to double check.

File: xml/System/String.xml [hunk 23]
```diff
@@ -11194,6 +11232,9 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
    
   
 ## Examples  
+
+ [!INCLUDE[interactive-note](~/includes/csharp-interactive-note.md)]
```
Comment (mairaw) [inline xml/System/String.xml]: you don't need this note here. You have already on top of the page.

```suggestion
```

File: xml/System/String.xml [hunk 24]
```diff
@@ -11194,6 +11232,9 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
+
```
Comment (mairaw) [inline xml/System/String.xml]:
```suggestion
```

File: xml/System/String.xml [hunk 25]
```diff
@@ -11481,6 +11522,9 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
    
   
 ## Examples  
+
+ [!INCLUDE[interactive-note](~/includes/csharp-interactive-note.md)]
```
Comment (mairaw) [inline xml/System/String.xml]: you don't need this note here. You have already on top of the page.

```suggestion
```

File: xml/System/String.xml [hunk 26]
```diff
@@ -11481,6 +11522,9 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
+
```
Comment (mairaw) [inline xml/System/String.xml]:
```suggestion
```

File: xml/System/String.xml [hunk 27]
```diff
@@ -12164,7 +12217,7 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
   
  The following example uses the <xref:System.String.Substring%2A> method to separate key/value pairs that are delimited by an equals ("=") character.
```
Comment (mairaw) [inline xml/System/String.xml]: the sample before this line seems that it would work with Try .NET as well and then the note could say all since it seems all samples would be runnable

File: xml/System/String.xml [hunk 28]
```diff
@@ -14136,10 +14197,13 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
    
   
 ## Examples  
+
+[!INCLUDE[interactive-note](~/includes/csharp-interactive-note.md)]
```
Comment (mairaw) [inline xml/System/String.xml]: you already have a note on top

```suggestion
```

File: xml/System/String.xml [hunk 29]
```diff
@@ -14136,10 +14197,13 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
    
   
 ## Examples  
+
+[!INCLUDE[interactive-note](~/includes/csharp-interactive-note.md)]  
+
```
Comment (mairaw) [inline xml/System/String.xml]:
```suggestion
```

**Comment by mairaw** [comment]
@WilliamAntonRohm any updates here?

File: xml/System/String.xml (hunk 19 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: @mairaw -- Maira, should I include the 16 remarks.cs snippets as interactive?  They have no output, but they can be played with.  Here's the first such snippet:
```
// System.String.Compare(System.String,System.Int32,System.String,System.Int32,System.Int32)
public class CompareSample1_1
{
    //<Snippet2>
    static bool IsFileURI(String path)
    {
        return (String.Compare(path, 0, "file:", 0, 5, true) == 0);
    }
    //</Snippet2>
}
```

File: xml/System/String.xml (hunk 22 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

File: xml/System/String.xml (hunk 23 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

File: xml/System/String.xml (hunk 25 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

File: xml/System/String.xml (hunk 28 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

File: xml/System/String.xml (hunk 29 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: @mairaw -- do you mean to delete blank line 14202 or the INCLUDE on line 14201?

**Comment by WilliamAntonRohm** [comment]
@mairaw -- I have been through Compare and updated the samples to match, adding three new interactive samples (comp3.cs [twice] and comp4.cs); see my note above regarding remarks.cs.  

Contains is good to go.

Some of the IndexOf example outputs do not match the Try .NET output, as the comparisons come back in Try .NET as -1, so I'm backing out any samples with culture comparisons:

```csharp
        string searchString = "\u00ADm";
        string s1 = "ani\u00ADmal" ;
        string s2 = "animal";

        Console.WriteLine(s1.IndexOf(searchString, 2, StringComparison.CurrentCulture));
        Console.WriteLine(s1.IndexOf(searchString, 2, StringComparison.Ordinal));
        Console.WriteLine(s2.IndexOf(searchString, 2, StringComparison.CurrentCulture));
        Console.WriteLine(s2.IndexOf(searchString, 2, StringComparison.Ordinal));

        // The example displays the following output:
        //       4
        //       3
        //       3
        //       -1
```

in Try .NET results in:

```
3
3
-1
-1
```

File: xml/System/String.xml (hunk 20 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

File: xml/System/String.xml (hunk 27 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

File: xml/System/String.xml (hunk 19 above)
Comment (mairaw) [reply xml/System/String.xml]: No @WilliamAntonRohm, just the ones with output

File: xml/System/String.xml (hunk 29 above)
Comment (mairaw) [reply xml/System/String.xml]: Usually when I leave suggestions you can just apply them directly on the GitHub UI (you can even batch them in the Files changed tab).

In this case, it was to remove the blank line.

**Comment by mairaw** [comment]
@WilliamAntonRohm I don't see any updates from today. Also, look at my comments for https://github.com/dotnet/samples/pull/1192 and see if you need to make any changes based on the changes I've made on top of that. Thanks!

File: xml/System/String.xml (hunk 29 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: That's what I thought, thanks for clarifying.  The GitHub UI is not allowing me to make these changes directly, but rather is giving me an error message; refreshing the page doesn't help.

**Comment by WilliamAntonRohm** [comment]
@mairaw -- I am hoping this build will succeed by morning.

Thank you for your close review of the latest samples, and for showing me their preferred formatting -- I'm checking the other post-interactivated samples.

**Comment by mairaw** [comment]
I didn't include that as part of the task description, so I'm just doing as I review the work.

File: xml/System/String.xml (hunk 21 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

**Comment by WilliamAntonRohm** [comment]
@mairaw -- Maira, I believe this new String.xml is ready to go live.  Thank you for your patience with my learning curve -- I anticipate the next ones (regex) will be much smoother.

**Comment by WilliamAntonRohm** [comment]
@mairaw -- I followed your master vs. upstream recommendations and just now created a replacement (new #3070) for this PR 2971, with only the one file changed (String.xml itself).  After your review, I'll make any changes in that new branch.

**Comment by mairaw** [comment]
We have new merge conflicts 😢

**Comment by WilliamAntonRohm** [comment]
@mairaw -- dang!  I'm unable to see the conflicts in this issue or its replacement https://github.com/dotnet/dotnet-api-docs/pull/3070, as the "Resolve conflicts" button is grayed-out, presumably a byproduct of my permissions level; I am likewise unable to assign an owner or reviewer.  If you could send me a raw text file of the conflicts I can re-create my additions.

**Comment by mairaw** [comment]
I can't see them either:
image

You need to use the command line to resolve this. If you need help, please let me know.
